rest/data/test.go

```diff
@@ -76,3 +81,61 @@ func (tc *DBTestConnector) FindTestsByTaskId(opts FindTestsByTaskIdOpts) ([]test
 
 	return res, nil
 }
+
+func TestCountByTaskID(ctx context.Context, taskID string, execution int) (int, error) {
+	t, err := task.FindOneIdOldOrNew(taskID, execution)
+	if err != nil {
+		return 0, gimlet.ErrorResponse{
+			StatusCode: http.StatusInternalServerError,
+			Message:    errors.Wrapf(err, "finding task '%s'", taskID).Error(),
+		}
+	}
+	if t == nil {
+		return 0, gimlet.ErrorResponse{
+			StatusCode: http.StatusNotFound,
+			Message:    fmt.Sprintf("task '%s' not found", taskID),
+		}
+	}
+
+	var count int
+	if t.HasCedarResults {
+		stats, status, err := apimodels.GetCedarTestResultsStats(ctx, apimodels.GetCedarTestResultsOptions{
+			BaseURL:     evergreen.GetEnvironment().Settings().Cedar.BaseURL,
+			TaskID:      taskID,
+			Execution:   utility.ToIntPtr(execution),
+			DisplayTask: t.DisplayOnly,
+		})
+		if err != nil {
+			return 0, gimlet.ErrorResponse{
+				StatusCode: http.StatusInternalServerError,
+				Message:    errors.Wrapf(err, "getting test results stats from Cedar for task '%s'", taskID).Error(),
+			}
+		}
+		if status != http.StatusOK && status != http.StatusNotFound {
+			return 0, gimlet.ErrorResponse{
+				StatusCode: http.StatusInternalServerError,
+				Message:    fmt.Sprintf("getting test results stats from Cedar for task '%s' returned status %d", taskID, status),
+			}
+		}
+
+		count = stats.TotalCount
```

Comment:
nit: rather than returning `count` at the very end of the entire if-else, would it be better to just return from this `if` block and unindent the else?

Reply:
Done.